phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java

```diff
@@ -135,8 +135,8 @@ private String getIndexedColumnsString(PTable indexPTable, PTable dataPTable, St
         StringBuilder indexedColumnsBuilder = new StringBuilder();
 
         for (PColumn indexedColumn : indexPK) {
-            String indexColumn = extractIndexColumn(indexedColumn.getName().getString(), defaultCF);
-            if(indexColumn.equalsIgnoreCase(MetaDataUtil.VIEW_INDEX_ID_COLUMN_NAME)) {
+            String indexColumn = extractIndexColumn(indexedColumn.getExpressionStr(), defaultCF);
+            if(indexColumn == null || indexColumn.equalsIgnoreCase(MetaDataUtil.VIEW_INDEX_ID_COLUMN_NAME)) {
```

Comment:
In what case do we get a null  here ?

Reply:
If indexedColumn.getExpressionStr() is null I return null in extractIndexColumn.

Reply:
And indexedColumn.getExpressionStr() is null when indexedColumn.getName().getString() was equal to MetaDataUtil.VIEW_INDEX_ID_COLUMN_NAME